lib/mariaex/protocol.ex

```diff
@@ -567,6 +572,9 @@ defmodule Mariaex.Protocol do
     end
   end
 
+  defp ping_handle(packet(msg: ok_resp()), :ping, %{buffer: buffer} = state) when is_binary(buffer) do
+    {:ok, state}
+  end
   defp ping_handle(packet(msg: ok_resp()), :ping, state) do
     activate(state, state.buffer)
   end
```

Comment:
`buffer` will always be binary, I think it would be clear to remove this activate and add
```elixir
{:ok, state} = ping_recv(state, :ping)
checkin(state)
```
to the `ping/1` clause where buffer is `:active_once`